control/control.go

```diff
@@ -1058,17 +1058,17 @@ func groupMetricTypesByPlugin(cat catalogsMetrics, metricTypes []core.Metric) (m
 		if err != nil {
 			return nil, serror.New(err)
 		}
-		returnedmt := plugin.PluginMetricType{
-			Namespace_:          incomingmt.Namespace(),
+		returnedmt := plugin.MetricType{
+			Namespace_:          catalogedmt.Namespace(),
 			LastAdvertisedTime_: catalogedmt.LastAdvertisedTime(),
 			Version_:            incomingmt.Version(),
 			Tags_:               catalogedmt.Tags(),
-			Labels_:             catalogedmt.Labels(),
 			Config_:             incomingmt.Config(),
+			Unit_:               catalogedmt.Unit(),
 		}
```

Comment:
I'm curious the reason we didn't define "metricType" as following. If that's the case, there is no need to set each field.  Is it because MetricType public and for the rest?

type metricType struct {
	Plugin             *loadedPlugin
	subscriptions      int
	policy             processesConfigData
	config             *cdata.ConfigDataNode
	metric             *plugin.MetricType
}